src/InstrumentationEngine.ProfilerProxy/ProxyLogging.h

```diff
@@ -39,6 +39,20 @@ namespace ProfilerProxy
     private:
         static HRESULT InitializeCore();
     };
+
+    // RAII for ProxyLogging
+    class CProxyLoggingHolder
+    {
+    public:
+        CProxyLoggingHolder()
+        {
+            CProxyLogging::Initialize();
```

Comment:
>CProxyLogging::Initialize(); [](start = 12, length = 28)

Have you tested this? I don't think that we have any test scenarios that initialize the logger multiple times. It should be possible to create a unit test for this.

Reply:
I've been testing my changes using src/scripts/windowslocaltest.ps1 (launches a powershell process with profiler variables). 

I'll create some unit tests to confirm this behavior.

---
In reply to: [365924547](https://github.com/microsoft/CLRInstrumentationEngine/pull/191#discussion_r365924547) [](ancestors = 365924547)

Reply:
Yeah. Unless your manual testing initializes multiple clrs in the same process at the same time, you will not have tested this code path completely.

---
In reply to: [365926752](https://github.com/microsoft/CLRInstrumentationEngine/pull/191#discussion_r365926752) [](ancestors = 365926752,365924547)

Reply:
I was able to test with a console app that calls DllGetClassObject twice and saw the eventlogger initialize and shutdown. I'll look into incorporating this console app as part of our test infra outside this PR.

---
In reply to: [365930535](https://github.com/microsoft/CLRInstrumentationEngine/pull/191#discussion_r365930535) [](ancestors = 365930535,365926752,365924547)